src/features/todo/todoSlice.jsx

```diff
@@ -0,0 +1,30 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const initialState = {
+    products: JSON.parse(localStorage.getItem('products')) || []
+}
+
+export const productSlice = createSlice({
+    name: 'products',
+    initialState,
+    reducers: {
+        addProduct: (state, action) => {
+            let found = false;
+            state.products.forEach((product) => {
+                if (product.info.name === action.payload.info.name) {
+                    product.number = (parseInt(product.number) || 0) + parseInt(action.payload.number);
```

Comment:
you [should not mutate state in redux](https://redux.js.org/faq/immutable-data#why-is-immutability-required-by-redux). here you mutate `number` property